pkgs/games/gargoyle/default.nix

```diff
@@ -1,68 +1,52 @@
-{ lib, stdenv, fetchFromGitHub, substituteAll, jam, cctools, pkg-config
-, SDL, SDL_mixer, SDL_sound, gtk2, libvorbis, smpeg }:
-
-let
-
-  jamenv = ''
-    unset AR
-  '' + (if stdenv.isDarwin then ''
-    export NIX_CFLAGS_COMPILE="$NIX_CFLAGS_COMPILE -I${lib.getDev SDL}/include/SDL"
-    export GARGLKINI="$out/Applications/Gargoyle.app/Contents/Resources/garglk.ini"
-  '' else ''
-    export NIX_LDFLAGS="$NIX_LDFLAGS -rpath $out/libexec/gargoyle"
-    export DESTDIR="$out"
-    export _BINDIR=libexec/gargoyle
-    export _APPDIR=libexec/gargoyle
-    export _LIBDIR=libexec/gargoyle
-    export GARGLKINI="$out/etc/garglk.ini"
-  '');
-
-in
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  substituteAll,
+  cctools,
+  pkg-config,
+  SDL2,
+  SDL2_mixer,
+  SDL2_sound,
+  qtbase,
+  qtwayland,
+  cmake,
+  wrapQtAppsHook,
+  libvorbis,
+  smpeg
+}:
 
 stdenv.mkDerivation rec {
   pname = "gargoyle";
-  version = "2019.1.1";
+  version = "2023.1";
 
   src = fetchFromGitHub {
     owner = "garglk";
     repo = "garglk";
     rev = version;
-    sha256 = "0w54avmbp4i4zps2rb4acmpa641s6wvwbrln4vbdhcz97fx48nzz";
+    hash = "sha256-XsN5FXWJb3DSOjipxr/HW9R7QS+7iEaITERTrbGEMwA=";
   };
 
-  nativeBuildInputs = [ jam pkg-config ] ++ lib.optional stdenv.isDarwin cctools;
-
-  buildInputs = [ SDL SDL_mixer SDL_sound gtk2 ]
-    ++ lib.optionals stdenv.isDarwin [ smpeg libvorbis ];
+  nativeBuildInputs = [ pkg-config cmake wrapQtAppsHook ] ++ lib.optional stdenv.isDarwin cctools;
 
+  buildInputs = [ SDL2 SDL2_mixer SDL2_sound qtbase ]
+    ++ lib.optionals stdenv.isDarwin [ smpeg libvorbis ]
+    ++ lib.optionals (!stdenv.isDarwin) [ qtwayland ];
+  cmakeFlags = [
+    "-DWITH_QT6=true"
+  ];
   # Workaround build failure on -fno-common toolchains:
   #   ld: build/linux.release/alan3/Location.o:(.bss+0x0): multiple definition of
   #     `logFile'; build/linux.release/alan3/act.o:(.bss+0x0): first defined here
   # TODO: drop once updated to 2022.1 or later.
-  env.NIX_CFLAGS_COMPILE = "-fcommon";
-
-  buildPhase = jamenv + "jam -j$NIX_BUILD_CORES";
+  # env.NIX_CFLAGS_COMPILE = "-fcommon";
 
-  installPhase =
-  if stdenv.isDarwin then
-  (substituteAll {
-    inherit (stdenv) shell;
-    isExecutable = true;
-    src = ./darwin.sh;
-  })
-  else jamenv + ''
-    jam -j$NIX_BUILD_CORES install
-    mkdir -p "$out/bin"
-    ln -s ../libexec/gargoyle/gargoyle "$out/bin"
-    mkdir -p "$out/etc"
-    cp garglk/garglk.ini "$out/etc"
-    mkdir -p "$out/share/applications"
-    cp garglk/gargoyle.desktop "$out/share/applications"
-    mkdir -p "$out/share/icons/hicolor/32x32/apps"
-    cp garglk/gargoyle-house.png "$out/share/icons/hicolor/32x32/apps"
+  # Fix the paths in .pc, even though it's unclear if these .pc are really useful.
+  postFixup = ''
+    substituteInPlace "$out"/lib/pkgconfig/garglk.pc \
+      --replace '=''${prefix}//' '='
```

Comment:
Thanks for having a look!

For now a fixup is needed to pass a check for `.pc` files with broken paths (https://github.com/NixOS/nixpkgs/pull/172347) (I don't entirely understand how the paths end up broken). I'll add a note that it can be removed in a future release.